src/app/shared/markdown-wrapper/markdown-wrapper.component.ts

```diff
@@ -0,0 +1,36 @@
+import { AfterViewInit, Component, Input, OnInit, ViewChild } from '@angular/core';
+import DOMPurify from 'dompurify';
+import { MarkdownComponent } from 'ngx-markdown';
+
+@Component({
+  selector: 'markdown-wrapper',
+  templateUrl: './markdown-wrapper.component.html',
+  styleUrls: ['./markdown-wrapper.component.scss'],
+})
+
+// Wrapper for user-input markdown to provide custom sanitization not accessible through the built-in sanitizer, DOMSanitize.
+// DOMSanitize's safe-lists can be seen https://github.com/angular/angular/blob/master/packages/core/src/sanitization/html_sanitizer.ts
+export class MarkdownWrapperComponent implements OnInit, AfterViewInit {
+  @ViewChild(MarkdownComponent) mkdRef;
+  @Input() data: string;
+  forbid_tags: string[];
+  forbid_attr: string[];
+
+  ngOnInit() {
+    // tags and attributes that will be sanitized by DOMPurify.
+    this.forbid_tags = [];
+    this.forbid_attr = ['class'];
+
+    // Sets config for DOMPurify until otherwise specified.
+    DOMPurify.setConfig({
+      FORBID_TAGS: this.forbid_tags,
+      FORBID_ATTR: this.forbid_attr,
+    });
+  }
+
+  ngAfterViewInit() {
+    // after the data has been passed through ngx-markdown, we can further sanitize the resulting HTML to our own
+    // specifications
+    this.mkdRef.element.nativeElement.innerHTML = DOMPurify.sanitize(this.mkdRef.element.nativeElement.innerHTML);
```

Comment:
Gary will know better, and you've tested, but I'm still slightly worried about accessing the DOM directly like this.

There may be no other way, but it's a general Angular best practice to not access the DOM directly.

Reply:
Updated. Now the input data variable is passed directly to HTML template with no need for manual DOM changes.